### PR TITLE
allow muxing private data

### PR DESCRIPTION
**Pull request by scottlamb**

### CODE REVIEW

File: mpegts-segmenter/src/segmenter.rs [hunk 1]
```diff
@@ -201,9 +205,7 @@ impl<S: SegmentStorage> Segmenter<S> {
                 }
 
                 if segment.temi_timeline_descriptor.is_none() {
-                    if let Some(af) = p.adaptation_field {
-                        segment.temi_timeline_descriptor = af.temi_timeline_descriptors.into_iter().next();
-                    }
+                    segment.temi_timeline_descriptor = first_temi_timeline_descriptor;
```
[the remark this answers is not in the document]
Comment (scottlamb) [reply mpegts-segmenter/src/segmenter.rs:208]: I don't understand your comment; can you rephrase?

I expect my code move to have no visible behavior change:

* if `p.adaptation_field` is `Some`, `first_temi_timeline_descriptor` has the same thing as this previous assignment. Notably, `p` hasn't changed since line 167.
* if `p.adaptation_field` is `None`, the assignment is a no-op because `segment.temi_timeline_descriptor` is `None` and so is `first_temi_timeline_descriptor`.

[the remark this answers is not in the document]
Comment (lzuosym) [reply mpegts-segmenter/src/segmenter.rs:208]: Sorry, my bad! the comment should be in line 176. I am afraid in some cases the `p.adaptation_field` is None, then `first_temi_timeline_descriptor` is reset to None.

[the remark this answers is not in the document]
Comment (scottlamb) [reply mpegts-segmenter/src/segmenter.rs:208]: `first_temi_timeline_descriptor` is never reset. It is bound on line 165 and assigned once in either line 167 or 176. 

If `first_temi_timeline_descriptor` is set to `None` in either place, then the assignment `segment.temi_timeline_descriptor = first_temi_timeline_descriptor` on line 208 is a no-op, because `segment.temi_timeline_descriptor` must also `None` or we wouldn't have taken the `if` statement on line 207.

[the remark this answers is not in the document]
Comment (lzuosym) [reply mpegts-segmenter/src/segmenter.rs:208]: Ah, I think you are right! In this case line 192 can change to
```diff
-                    temi_timeline_descriptor: None,
+                    temi_timeline_descriptor: first_temi_timeline_descriptor,
```

But I still like the old way without the need of additional `first_temi_timeline_descriptor` variable, more straightforward.

**Comment by lzuosym** [comment]
The CI build seems to be stuck on unit test. It's been running for more than 3 hours.

File: mpegts-segmenter/src/segmenter.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (scottlamb) [reply mpegts-segmenter/src/segmenter.rs:208]: > Ah, I think you are right! In this case line 192 can change to

Wouldn't that change to line 192 be redundant with the assignment on line 208? Removing the latter would be a behavior change in the case where the segment's first packet doesn't have a descriptor but a later one does.

> But I still like the old way without the need of additional `first_temi_timeline_descriptor` variable, more straightforward.

I would like this to be straightforward to read, without having to do unnecessary clones/copies. Would renaming the variable help? `p_first_temi_timeline_descriptor` maybe to emphasize it's the first within `p` rather than the first of the stream/segment/whatever?

> The CI build seems to be stuck on unit test. It's been running for more than 3 hours.

Sigh. Canceled, started a rerun, fingers crossed.

**Review by lzuosym — APPROVED** [review]
